native/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -0,0 +1,252 @@
+import org.ballerinalang.formatter.core.FormatterException;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+/**
+ * Class to implement "persist" command for ballerina.
+ */
+@CommandLine.Command(
+        name = "generate",
+        description = "generate database configurations."
+        )
+
+public class Generate implements BLauncherCmd {
+
+    private static final PrintStream errStream = System.err;
+
+    private String sourcePath = "";
+    public ProjectEnvironmentBuilder projectEnvironmentBuilder;
+
+    private static final String COMMAND_IDENTIFIER = "persist-generate";
+
+    private CommandLine parentCmdParser;
+
+    Project balProject;
+
+    String name;
+
+    @CommandLine.Option(names = {"-h", "--help"}, hidden = true)
+    private boolean helpFlag;
+
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
+            errStream.println(commandUsageInfo);
+            return;
+        }
+        try  {
+            if (projectEnvironmentBuilder == null) {
+                balProject = ProjectLoader.loadProject(Paths.get(""));
+            } else {
+                balProject = ProjectLoader.loadProject(Paths.get(this.sourcePath), projectEnvironmentBuilder);
+            }
+            name = balProject.currentPackage().descriptor().org().value() + "." + balProject.currentPackage()
+                    .descriptor().name().value();
+
+
+        } catch (ProjectException e) {
+            errStream.println("The current directory is not a Ballerina project!");
+            return;
+        }
+        try {
+            ArrayList<Entity> entityArray = readBalFiles();
+            if (entityArray.size() == 0) {
+                errStream.println("Current directory doesn't contain any persist entities");
+            } else {
+                for (Entity entity : entityArray) {
+                    entity.setPackageName(balProject.currentPackage().descriptor().org().value() + "/"
+                            + balProject.currentPackage().descriptor().name().value());
+                    generateClientBalFile(entity);
+                }
+                generateConfigurationBalFile();
+            }
+        } catch (Exception e) {
+            errStream.println(e.getMessage());
+        }
+    }
+
+    private ArrayList<Entity> readBalFiles() throws BalException {
+        ArrayList<Entity> returnMetaData = new ArrayList<>();
+        Path dirPath = Paths.get(this.sourcePath);
+        List<Path> fileList;
+
+
```

Comment:
```suggestion

```

Reply:
Updated in [895edde](https://github.com/ballerina-platform/persist-tools/pull/15/commits/895eddeb28babc700ff8c472fe1745b73ebe9286)